include/emds/emds.h

```diff
@@ -35,6 +48,11 @@ struct emds_entry {
 	uint8_t *data;
 	/** Length of data that will be stored. */
 	size_t len;
+	/** Entry status. */
+	union {
+		enum emds_entry_status val;
+		enum emds_entry_status *ptr;
```

Comment:
Do we really need both val and *ptr?

Reply:
I investigated it a bit. Seems like all members in the static entries are constants, meaning that it is not possible to alter it after declaration. I thought this might be the best option since the alternative is to just keep the pointer , which means that we would have to initialize every dynamic entry with a pointer to a status instance.